apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java

```diff
@@ -462,6 +463,17 @@ public boolean stopSpan(AbstractSpan span) {
         finish();
     }
 
+    @Override
+    public void prepareProfiling(String firstSpanOPName) {
+        if (segment.getProfiling()) {
+            return;
+        }
+
+        // update profiling status
+        final ProfileTaskExecutionService profileTaskExecutionService = ServiceManager.INSTANCE.findService(ProfileTaskExecutionService.class);
+        segment.setProfiling(profileTaskExecutionService.addProfiling(segment, firstSpanOPName));
```

Comment:
You should use `TracingContext` rather than `segment` as the parameter. `segment` is a collection of spans, only tracingContext has status.